scripts/migrations/postgres/20211109185301_referral.up.sql

```diff
@@ -86,4 +86,36 @@ CREATE TABLE referral_tracking (
     sender_reward INT,
     receiver_reward INT,
     PRIMARY KEY (sender, receiver)
-);
\ No newline at end of file
+);
+
+CREATE INDEX referral_tracking_registered_at_idx ON referral_tracking(registered_at);
+
+CREATE OR REPLACE FUNCTION referral_tracking_sender_stats(addr VARCHAR, since INTERVAL)
+RETURNS TABLE (
+    registered INT,
+    installed INT,
+    confirmed INT,
+    reward INT
+) AS $$
+BEGIN
+    RETURN QUERY
+        SELECT COALESCE(
+                    (SELECT COUNT (*) FROM referral_tracking
+                    WHERE sender = addr AND
+                          CASE WHEN since IS NULL THEN TRUE ELSE registered_at > NOW() - since END), 0)::INT AS registered,
+               COALESCE(
+                    (SELECT COUNT (*) FROM referral_tracking
+                    WHERE sender = addr AND installed_at IS NOT NULL AND
+                          CASE WHEN since IS NULL THEN TRUE ELSE registered_at > NOW() - since END), 0)::INT AS installed,
+               COALESCE(
+                   (SELECT COUNT (*) FROM referral_tracking
+                    WHERE sender = addr AND confirmed_at IS NOT NULL AND
+                          CASE WHEN since IS NULL THEN TRUE ELSE registered_at > NOW() - since END), 0)::INT AS confirmed,
+               COALESCE(
+                   (SELECT SUM (COALESCE(sender_reward, 0)) FROM referral_tracking
+                   WHERE sender = addr AND
+                         CASE WHEN since IS NULL THEN TRUE ELSE registered_at > NOW() - since END), 0)::INT AS reward;
```

Comment:
It would be great to format the code a bit

Reply:
also what do you think about to add `f_` prefix to the function's name?

Reply:
formatted.
Regarding `_f` I don't think we need it. There are no places with context when you are not able to identify where it is a func name or not.